### PR TITLE
feature/add-catch-all-route

### LINKED ISSUE #102: Add catchall route
### Feature Description

_A clear and concise description of the feature you're requesting._

In the case an event, person, or just `/*/` doesn't exist, show a page that says so

### Use Case

_Please provide a use case to help us understand your request in context._

Better user experience to know something is wrong than simply just having no content load.

### Solution

_Please describe your ideal solution._

Some sort of "Content not found", look at mozilla: https://www.mozilla.org/en-US/example-does-not-exist
image

### PR DESCRIPTION
**Pull request by NeelApte**
### Link to Relevant Issue

This pull request resolves #102 

### Description of Changes

Displays an error message when the route given is not provided. Also provides a general error page for future development. 

### Link to Site

https://neelapte.github.io/cdp-frontend/#/9999

This page displays a page not found error. 

https://neelapte.github.io/cdp-frontend/#/events/8888

This page displays a formatted error. 

_Please see `CONTRIBUTING.md` for directions on how this can be done._

### CODE REVIEW

File: src/app/App.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import { EventPage } from "../pages/EventPage";
 import { EventsPage } from "../pages/EventsPage";
 import { PersonPage } from "../pages/PersonPage";
 import { PeoplePage } from "../pages/PeoplePage";
+import ErrorPage from "../pages/ErrorPage/ErrorPage";
```
Comment (tohuynh) [inline src/app/App.tsx:19]: Change to
```
import { ErrorPage } from "../pages/ErrorPage";
```

for consistent imports

File: src/app/App.tsx [hunk 2]
```diff
@@ -101,6 +102,7 @@ function App() {
                 <Route exact path="/people/:id">
                   <PersonPage />
                 </Route>
+                <Route component={ErrorPage} />
```
Comment (tohuynh) [inline src/app/App.tsx:105]: Add a `<Route path="*" component={ErrorPage} />` as recommended by the [react-router doc](https://v5.reactrouter.com/web/example/no-match).

File: src/pages/ErrorPage/ErrorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+const ErrorPage: FC<ErrorPageProps> = ({ error }: ErrorPageProps) => {
+  const history = useHistory();
+  const errorText = error ? error.toString() : "Sorry, we can’t find that page.";
+
+  return (
+    <div>
+      <h1>{errorText}</h1>
+      <button onClick={() => history.goBack()}>Back</button>
```
Comment (tohuynh) [inline src/pages/ErrorPage/ErrorPage.tsx:15]: I think the content of h1 should always be "Sorry, we can’t find that page." The h1 should have className="mzp-u-title-sm" to be consistent with other page titles.

Then the error message (if there is an error message) can be displayed below the title inside a `p`.

Also, we can use one of the [Mozilla protocol buttons](https://protocol.mozilla.org/patterns/atoms/buttons.html). Neutral or secondary styles look good to me.

**Review by tohuynh — CHANGES_REQUESTED** [review]
Looks good. I have few minor asks for the styling.

Just a heads up regarding git history. Your local `main` should contain only peer-reviewed code. Meaning you shouldn't merge local changes to your local `main` branch. Instead, you can get the latest changes from upstream (this repo)'s `main` branch if you need to resolve a merge conflict. This way you can keep the commit history of the PR clean (e.g. it wouldn't include your Underline commit in this PR).

**Comment by BrianL3** [comment]
Great suggestions To. Neel, super pleased to see your work 👍

**Review by BrianL3 — APPROVED** [review]
I paired with Neel this afternoon!  Great fun, glad he's on the project.

**Review by JacksonMaxfield — COMMENTED** [review]
So happy this is in! Agree with To that we should try to use a mozilla styled button but otherwise. I am pretty happy with this. I leave code comments to To and Brian

**Review by tohuynh — APPROVED** [review]
Thanks for making the changes.

**Pull request merged**